### PR TITLE
Assignment 1 completed and tested

### PR DESCRIPTION
**Pull request by darlingoscanoa**
UofT-DSI | Deep Learning - Assignment 1

## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
I completed the Lab 1 and Lab 2 and also the Assingment 1 following the instructions given for the professor during class.

## What did you learn from the changes you have made?
First I learn how easy is to use Tensorflow by defining the 3 main steps in each architecture: build the model, compile, train
After learning how the back and forward propagation works was easier to then implement ANN
Then after working with images in the assignment 1 we learn how to use CNN, an specific ANN for image dataset to learn about their patterns like edge, texture and so on.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Other approach should be difficult to learn for these image dataset classification tasks

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Yes, my visual code exploit! and I need to move to Google Colab to complete the assignment

## How were these changes tested?
We review the improvement of the accuracy in each step after making the changes

## A reference to a related issue in your repository (if applicable)
-
## Checklist
- [x ] I can confirm that my changes are working as intended

### CODE REVIEW

File: 01_materials/labs/lab_2.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_2.ipynb:1]: **Hyperparameters settings**: You implemented a neural network with two hidden layers and achieved worse accuracy compared to the previous baseline model. I noticed that you used the `sigmoid()` activation for the first layer. `sigmoid()` takes a real number and outputs a value in the range of `[0,1]`, which may cause [gradient vanishing](https://www.kdnuggets.com/2022/02/vanishing-gradient-problem.html) issue in larger models. I re-ran your code by replacing `sigmoid()` with `relu()` and doubling the training epochs, and got ~90% accuracy.

File: 02_activities/assignments/assignment_1.ipynb
Comment (tianyi21) [inline 02_activities/assignments/assignment_1.ipynb:1]: **3. Building and Evaluating a Simple CNN Model**: the question states that you are supposed to implement `at least one convolutional layer (to learn spatial hierarchies in images) and one fully connected layer`, so a `MaxPool` layer is not required here.

**Review by tianyi21 — APPROVED** [review]
Hi Darling,

Thank you for submitting your DL week 1 assessments.

Please find my comments attached to each file. You A1, Lab 1, and Lab 2 are now complete.

Thanks,
Tianyi [LS]
